### ISSUE TITLE
specWarnGTFO reuse for multiple spells shows wrong icon

### ISSUE THREAD

**Issue by Zidras** [DeadlyBossMods/DBM-Unified#267]
**Describe the bug**
GTFO is being commonly re-used for multiple spells inside the same mod, which leads to the icon being wrong if it doesn't match the metadata set in mod load.

**Do you have an error log of what happened?**
NA

**To Reproduce**
Example:
https://github.com/DeadlyBossMods/DBM-WotLK/blob/1e1803885425d59091953699422c987366d8fb7f/DBM-Raids-WoTLK/Coliseum/NorthrendBeasts.lua#L254

**Screenshots**
<img width="482" alt="image" src="https://github.com/DeadlyBossMods/DBM-Unified/assets/10605951/f2084eed-4127-4bc2-bd06-93c96d83d108">

**Did you try having DeadlyBossMods as the only enabled addon and everything else (especially something like ElvUI) disabled?**
Yes

**Which version of DeadlyBossMods are you using?**
Independent of version

**Was it working in a previous version? If yes, which was the last good one?**
No

**Additional context**
There are also several cases (on multiple mods) where the text argument for the %s locale string is not passed:
https://github.com/DeadlyBossMods/DBM-Unified/blob/master/DBM-Core/localization.en.lua#L400
One example, with two different spells and different icons, and no spellname passed (which defaults to "bad" damage and wrong icon):
https://github.com/DeadlyBossMods/DBM-WotLK/blob/1e1803885425d59091953699422c987366d8fb7f/DBM-Raids-WoTLK/Coliseum/Jaraxxus.lua#L158-L159
![image](https://github.com/DeadlyBossMods/DBM-Unified/assets/10605951/a4075686-9b64-4d55-8103-12ab983c8093)
This might be intended, but decided to write it here as well since retail mods all pass the spellname into gtfo Show.

**Comment by MysticalOS** [comment]
Yeah I know, it's an inconsequential issue really as fixing it would require annoying effort of calling seticon ever time it's reused in every mod fore very spell. passing spellname though is something that is usually done though so a mod not doing that is worth fixing but that's about it

**Comment by MysticalOS** [comment]
Fixed 3 spells in wraith raids to now show spell name

**Comment by Zidras** [comment]
Why not use `UpdateKey` before every gtfo Show with multiple id? Is there anything wrong with it?
https://github.com/DeadlyBossMods/DBM-Unified/blob/master/DBM-Core/DBM-Core.lua#L9299

With this method, you can quickly update the icon on the fly:
```lua
function mod:SPELL_DAMAGE(_, _, _, _, destGUID, _, _, _, spellId, spellName)
	if (spellId == 66317 or spellId == 66320 or spellId == 66881) and destGUID == UnitGUID("player") and self:AntiSpam(3, 1) then	-- Fire Bomb (66317 is impact damage, not avoidable but leaving in because it still means earliest possible warning to move. Other 4 are tick damage from standing in it)
		specWarnGTFO:UpdateKey(spellId) -- added
		specWarnGTFO:Show(spellName)
		specWarnGTFO:Play("watchfeet")
	end
end
```
Might not be the cleanest implementation due to how gtfo is widespread and not unique, but is a low effort hotfix, don't you agree? At least for your future mods this is something that can be used.
